Firebase/android/app/src/main/java/com/novoda/bonfire/channel/view/NewChannelView.java

```diff
@@ -99,17 +105,30 @@ private boolean isEmojiComponent(char c) {
 
         @Override
         public void afterTextChanged(Editable s) {
-            createButton.setEnabled(s.length() > 0 && isValidInput);
+            toolbar.getMenu().getItem(0).setEnabled(s.length() > 0 && isValidInput);
             if (!isValidInput) {
                 setChannelNameError(R.string.only_single_emoji_allowed);
             }
         }
     };
 
-    private final OnClickListener createButtonClickListener = new OnClickListener() {
+    private final Toolbar.OnMenuItemClickListener menuItemClickListener = new Toolbar.OnMenuItemClickListener() {
+        @Override
+        public boolean onMenuItemClick(MenuItem item) {
+            switch (item.getItemId()) {
```

Comment:
Do you plan on adding more stuff to the menu? if not, I'd go with a plain old `if`.